### PR TITLE
[fix](multi-catalog)support oss-hdfs service

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

1. support oss-hdfs if it is enabled when use dlf or hms catalog
2. add docs for aliyun dlf and mc.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
Now I can not find any method to get the jindosdk mvn dependencies.

**Comment by wsjz** [comment]
Now just support manual download from  https://github.com/aliyun/alibabacloud-jindodata/blob/master/docs/user/5.x/5.0.0-beta7/jindodata_download.md

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/common/util/S3Util.java [hunk 1]
```diff
@@ -87,11 +87,20 @@ public static Path toScanRangeLocation(String location) {
...
+                location = "oss" + location.substring(pos);
+            } else {
+                location = "s3" + location.substring(pos);
+            }
         }
         return new Path(location);
     }
 
+    public static boolean isHdfsOnOssEndpoint(String location) {
+        return location.contains("oss-dls.aliyuncs");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/S3Util.java:101]: Add reference link

File: fe/fe-core/src/main/java/org/apache/doris/common/util/S3Util.java [hunk 2]
```diff
@@ -87,11 +87,20 @@ public static Path toScanRangeLocation(String location) {
             if (pos == -1) {
                 throw new RuntimeException("No '://' found in location: " + location);
             }
-            location = "s3" + location.substring(pos);
+            if (isHdfsOnOssEndpoint(location)) {
+                // if hdfs service is enabled on oss, use oss location
+                location = "oss" + location.substring(pos);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/S3Util.java:92]: Add example

File: fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java [hunk 3]
```diff
@@ -268,6 +274,21 @@ private static Map<String, String> convertToOSSProperties(Map<String, String> pr
         return ossProperties;
     }
 
+    private static void rewriteHdfsOnOssProperties(Map<String, String> ossProperties, String endpoint) {
+        if (!S3Util.isHdfsOnOssEndpoint(endpoint)) {
+            // just for robustness here, avoid wrong endpoint when oss-hdfs is enabled.
+            // convert "oss-cn-beijing.aliyuncs.com" to "cn-beijing.oss-dls.aliyuncs.com"
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java:280]: Add reference link

**Comment by wsjz** [comment]
run buildall

File: docs/en/docs/lakehouse/multi-catalog/dlf.md [hunk 4]
```diff
@@ -35,83 +35,72 @@ Doris can access DLF the same way as it accesses Hive Metastore.
 
 ## Connect to DLF
 
-### The First Way, Create a Hive Catalog.
+### Create a DLF Catalog.
 
 ```sql
-CREATE CATALOG hive_with_dlf PROPERTIES (
+CREATE CATALOG _dlf PROPERTIES (
```
Comment (morningman) [inline docs/en/docs/lakehouse/multi-catalog/dlf.md:41]:
```suggestion
CREATE CATALOG dlf PROPERTIES (
```

File: docs/zh-CN/docs/lakehouse/multi-catalog/max_compute.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (morningman) [inline docs/zh-CN/docs/lakehouse/multi-catalog/max_compute.md:1]: Add this to sidebar

File: docs/zh-CN/docs/lakehouse/multi-catalog/max_compute.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+
+# 阿里云 MaxCompute
```
Comment (morningman) [inline docs/zh-CN/docs/lakehouse/multi-catalog/max_compute.md:28]: 需要说明下限额的事情

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 49.7 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162669476 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230713005243_clickbench_pr_177527.html

**Comment by wsjz** [comment]
run fe ut

**Comment by wsjz** [comment]
run feut

**Comment by wsjz** [comment]
run ut

**Comment by wsjz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.28 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17163283811 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230713105119_clickbench_pr_177660.html

**Review by morningman — APPROVED** [review]
LGTM

**Comment by wsjz** [comment]
run p0

**Review by zddr — APPROVED** [review]

**Pull request merged**